lib/middlewares.ts

```diff
@@ -32,3 +36,31 @@ export const webhookMiddleware = (
   domainMiddleware(request);
   eventMiddleware(request, expectedEvent);
 };
+
+export const jwtVerifyMiddleware = async (request: NextApiRequest) => {
+  const {
+    [Constants.SALEOR_DOMAIN_HEADER]: domain,
+    "authorization-bearer": token
+  } = request.headers;
+  const tokenClaims = jwt.decode(token as string);
+
+  if (tokenClaims === null) {
+    throw new MiddlewareError("Missing token.", 400);
+  }
+  if ((tokenClaims as JwtPayload).iss !== domain) {
+    throw new MiddlewareError("Invalid token.", 400);
+  }
+
+  const jwksClient = jwks({ jwksUri: `https://${domain}/.well-known/jwks.json` });
+  const signingKey = await jwksClient.getSigningKey();
+  const signingSecret =
+    (signingKey as CertSigningKey).publicKey ||
+    (signingKey as RsaSigningKey).rsaPublicKey;
+
+  try {
+    jwt.verify(token as string, signingSecret);
+  } catch (e) {
+    console.log(e);
```

Comment:
```suggestion
    console.error(e);
```
Let's keep it consistent with webhook try..catch